R/radqlqc.R

```diff
@@ -0,0 +1,1304 @@
+    ) %>%
+    group_by(
+      USUBJID,
+      ADTM
+    ) %>%
+    summarise(first_date = first(ADTM), .groups = "drop")
+
+  ADQLQC <- left_join(
+    ADQLQC,
+    ADQLQC_x
+  ) %>%
+    mutate(
+      ANL01FL = case_when(
+        PARAMCD != "QSALL" & ABLFL == "Y" ~ "Y",
+        PARAMCD != "QSALL" &
+          !str_detect(AVISIT, "UNSCHEDULED") &
+          !is.na(first_date) ~ "Y"
+      )
+    ) %>%
+    select(-first_date)
+
+  # final dataset -----------------------------------------------------------
+  ADQLQC_final <- ADQLQC %>% # nolint
+    dplyr::group_by(.data$USUBJID) %>%
+    dplyr::mutate(ASEQ = row_number()) %>%
+    dplyr::ungroup() %>%
+    dplyr::arrange(
+      .data$STUDYID,
+      .data$USUBJID,
+      .data$AVISITN
+    ) %>%
+    select(
+      -c("BASE2", "CHG2", "PCHG2", "ABLFL2")
+    ) %>%
+    ungroup()
+
+  # ADQLQC variables from gdsr
+  adam_vars <- adamv_variables() %>%
+    filter(.data$dataset == "ADQLQC")
+
+  # ordered variables as per gdsr
+  adam_vars <- adam_vars$name
+
+  # order variables in mapped qs by variables in adam_vars
+  adqlqc_name_ordered <- names(ADQLQC_final)[order(match(names(ADQLQC_final), adam_vars))]
+
+  # adqlqc with variables ordered per gdsr
+  ADQLQC_final <- ADQLQC_final %>%
+    select(
+      any_of(adqlqc_name_ordered)
+    )
+
+  ADQLQC_final <- relocate(ADQLQC_final, "QSEVLINT", .after = "QSTESTCD") %>%
+    arrange(
+      .data$USUBJID,
+      .data$AVISITN,
+      .data$ASEQ,
+      .data$QSTESTCD
+    )
+
+  # apply metadata
+  ADQLQC_final <- random.cdisc.data:::apply_metadata(ADQLQC_final, "metadata/ADQLQC.yml") # nolint
+
+  return(ADQLQC_final)
+}
+
+#' Questionnaires EORTC QLQ-C30 V3.0 SDTM (QS)
+#' Function for generating random Questionnaires SDTM domain
+#' @param visit_format as character string. Valid values: WEEK, CYCLE.
+#' @param n_assessments number of assessments. Valid values: integer.
+#' @param n_days number of days for each cycle: Valid values: integer.
+#' @param lookup control lookup process.
+#' @param seed Seed for random number generation.
+#' @param na_percentage (\code{numeric}) Default percentage of values to be replaced by NA
+#' @param na_vars na_vars (\code{list}) A named list where the name of each element is a column name of \code{ds}. Each
+#' element of this list should be a numeric vector with two elements
+#'  \itemize{
+#'     \item{seed }{The seed to be used for this element - can be left NA}
+#'     \item{percentage }{How many element should be replaced. 0 is 0 \% 1 is 100 \%, can be left NA and default
+#'     percentage is used. This will overwrite default percentage (percentage argument))}
+#' }
+#' @param cached boolean whether the cached <%= toupper(data) %> data \code{c<%=data%>} should be
+#' returned or new data
+#'
+#' @return a dataframe with SDTM questionnaire data
+#'
+#' @import random.cdisc.data
+#' @import dplyr
+#' @importFrom rgdsr gdsr_init sdtmv_variables
+#'
+#' @export
```

Comment:
```suggestion
```

We can make this function as internal. no need to export

Reply:
Thank you @shajoezhu, export removed